bitnami/mongodb/templates/hidden/statefulset.yaml

```diff
@@ -0,0 +1,480 @@
+        nodeAffinity: {{- include "common.affinities.nodes" (dict "type" .Values.hidden.nodeAffinityPreset.type "key" .Values.nodeAffinityPreset.key "values" .Values.nodeAffinityPreset.values) | nindent 10 }}
+      {{- end }}
+      {{- if .Values.hidden.nodeSelector }}
+      nodeSelector: {{- include "common.tplvalues.render" (dict "value" .Values.hidden.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.hidden.tolerations }}
+      tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.hidden.tolerations "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.hidden.priorityClassName }}
+      priorityClassName: {{ .Values.hidden.priorityClassName }}
+      {{- end }}
+      {{- if .Values.hidden.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.hidden.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if or .Values.hidden.initContainers (and .Values.hidden.volumePermissions.enabled .Values.hidden.persistence.enabled) (and .Values.hidden.externalAccess.enabled .Values.hidden.externalAccess.autoDiscovery.enabled) .Values.tls.enabled }}
+      initContainers:
+        {{- if .Values.hidden.initContainers }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.hidden.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+        {{- if and .Values.hidden.volumePermissions.enabled .Values.hidden.persistence.enabled }}
+        - name: volume-permissions
+          image: {{ include "mongodb.volumePermissions.image" . }}
+          imagePullPolicy: {{ .Values.hidden.volumePermissions.image.pullPolicy | quote }}
+          command:
+            - /bin/bash
+            - -ec
+          args:
+            - |
+              mkdir -p {{ .Values.hidden.persistence.mountPath }}
+              {{- if and .Values.hidden.podSecurityContext.enabled .Values.hidden.containerSecurityContext.enabled }}
+              chown -R "{{ .Values.hidden.containerSecurityContext.runAsUser }}:{{ .Values.hidden.podSecurityContext.fsGroup }}" "{{ .Values.hidden.persistence.mountPath }}"
+              {{- end }}
+          {{- if eq ( toString ( .Values.hidden.volumePermissions.securityContext.runAsUser )) "auto" }}
+          securityContext: {{- omit .Values.hidden.volumePermissions.securityContext "runAsUser" | toYaml | nindent 12 }}
+          {{- else }}
+          securityContext: {{- .Values.hidden.volumePermissions.securityContext | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.hidden.volumePermissions.resources }}
+          resources: {{- toYaml .Values.hidden.volumePermissions.resources | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: datadir
+              mountPath: {{ .Values.hidden.persistence.mountPath }}
+        {{- end }}
```

Comment:
I'm not sure hidden nodes need their own `hidden.volumePermissions.*` and `hidden.containerSecurityContext.*` configuration.

I think it should be using the same as the other MongoDB nodes to prevent accidental misconfiguration by having duplicated config.

Also in line 79 it will load the image configured in `.Values.VolumePermissions` and not the image configured in `.Values.hidden.volumePermissions`

Reply:
I agree we should use the same, Updated.